spec-0007/transition_to_rng.py

```diff
@@ -0,0 +1,157 @@
+import numpy as np
+import functools
+import warnings
+
+
+def _transition_to_rng(old_name, position_num=None, dep_version=None):
+    """Example decorator to transition from old PRNG usage to new `rng` behavior
+
+    Suppose the decorator is applied to a function that used to accept parameter
+    `old_name='random_state'` either by keyword or as a positional argument at
+    `position_name=1`. At the time of application, the name of the argument in the
+    function signature is manually changed to the new name, `rng`. If positional
+    use was allowed before, this is not changed.*
+
+    - If the function is called with both `random_state` and `rng`, the decorator
+      raises an error.
+    - If `random_state` is provided as a keyword argument, the decorator passes
+      `random_state` to the function's `rng` argument as a keyword, but the decorator
+      will emit a `DeprecationWarning` about the deprecation of keyword `random_state`.
+    - If `random_state` is provided as a positional argument, the decorator passes
+      `random_state` to the function's `rng` argument by position, but the decorator
+      will emit a `FutureWarning` about the changing interpretation of the argument.
+    - If `rng` is provided as a keyword argument, the decorator validates `rng` using
+      `numpy.random.default_rng` before passing it to the function.
+    - If neither `random_state` nor `rng` is provided, the decorator checks whether
+      `np.random.seed` has been used to set the global seed. If so, it emits a
+      `FutureWarning`, noting that usage of `numpy.random.seed` will eventually have
+      no effect. Either way, the decorator calls the function without explicitly
+      passing the `rng` argument.
+
+    To avoid warnings, a user must pass `rng` as a keyword or pass a `Generator`
+    object or `None` by position.
```

Comment:
```suggestion
    To avoid warnings, a user must pass `rng` as a keyword.
```
Oops, we discussed this but I forgot to remove it.